### PR TITLE
feat(InputManager): Switch to using udev for device discovery

### PR DESCRIPTION
**Pull request by pastaq**

### CODE REVIEW

File: src/config/mod.rs [hunk 1]
```diff
@@ -385,59 +382,78 @@ impl CompositeDeviceConfig {
...
+            if vid != vendor_id {
                 return false;
             }
         }
 
         if let Some(product_id) = hidraw_config.product_id {
-            if device.product_id() != product_id {
+            let pid = device.id_product();
+            log::trace!("Checking product_id: {product_id} against {pid}");
+            if pid != device.id_product() {
```
Comment (ShadowApex) [inline src/config/mod.rs:440]:
```suggestion
            if pid != product_id {
```

File: src/input/manager.rs [hunk 2]
```diff
@@ -707,7 +662,7 @@ impl Manager {
                                     if let Some(ignored) = source_device.ignore {
                                         if ignored {
                                             log::debug!("Ignoring device {:?}, not adding to composite device: {}", source_device, composite_device);
-                                            continue;
+                                            break 'start;
```
[the remark this answers is not in the document]
Comment (pastaq) [reply src/input/manager.rs:665]: This hits every loop. We didn't need to keep checking if we know the device is ignored. Noticed it during testing.

File: src/input/composite_device/mod.rs [hunk 3]
```diff
@@ -1271,49 +1265,18 @@ impl CompositeDevice {
...
-        if id.starts_with("evdev://") {
-            let name = id.strip_prefix("evdev://").unwrap();
-            let path = format!("/dev/input/{}", name);
-
-            if let Some(idx) = self.source_device_paths.iter().position(|str| str == &path) {
-                self.source_device_paths.remove(idx);
-            };
+    async fn on_source_device_removed(&mut self, device: UdevDevice) -> Result<(), Box<dyn Error>> {
+        let path = device.devpath();
+        let id = device.devnode();
```
[the remark this answers is not in the document]
Comment (pastaq) [reply src/input/composite_device/mod.rs:1270]: Possibly. We may want to store those properties on the udevdevice. I'll do some more testing.

File: src/input/manager.rs [hunk 4]
```diff
@@ -1204,7 +1204,9 @@ impl Manager {
 
     async fn on_device_removed(&mut self, device: UdevDevice) -> Result<(), Box<dyn Error>> {
         log::debug!("Device removed: {:?}", device.name());
-        let path = ObjectPath::from_string_unchecked(device.devpath());
+        let sys_name = device.syspath.split('/').last().unwrap();
+        let path =
+            ObjectPath::from_string_unchecked(format!("{BUS_PREFIX}/devices/source/{sys_name}"));
```
Comment (pastaq) [inline src/input/manager.rs:1209]: We can use device.sysname() instead of splitting the syspath.

Any reason to not use BUS_SOURCE_PREFIX instead?

**Review by ShadowApex — APPROVED** [review]

**Pull request merged**